src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.cs

```diff
@@ -12,74 +12,81 @@ namespace Microsoft.Identity.Abstractions
         /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
-        Task<HttpResponseMessage> CallRestApiForUserAsync(
+        public Task<HttpResponseMessage> CallRestApiAsync(
             string serviceName,
-            Action<DownstreamRestApiOptions>? calledDownstreamRestApiOptionsOverride = null,
-            ClaimsPrincipal? user = default,
+            Action<DownstreamRestApiOptions>? downstreamRestApiOptionsOverride = null,
+            ClaimsPrincipal? user = null,
             HttpContent? content = null,
             CancellationToken cancellationToken = default);
 
+
         /// <summary>
-        /// Calls the downstream REST API on behalf of the app itself, with the required scopes.
+        /// Calls the downstream REST API on behalf of the user, based on a description of the
+        /// downstream REST API in the configuration (service name), overridatable programmatically. This
+        /// is a lower level API. There are other APIs for specific Http methods.
         /// </summary>
-        /// <param name="serviceName">Name of the service describing the downstream REST API. There can
+        /// <param name="serviceName">Name of the service describing the downstream web API. There can
         /// be several configuration named sections mapped to a <see cref="DownstreamRestApiOptions"/>,
-        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="DownstreamRestApiOptionsOverride"/>
+        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="downstreamRestApiOptionsOverride"/>
         /// needs to be set.</param>
-        /// <param name="DownstreamRestApiOptionsOverride">(Optional) Overrides the options proposed in the configuration described
+        /// <param name="downstreamRestApiOptionsOverride">(Optional) Overrides the options proposed in the configuration described
         /// by <paramref name="serviceName"/>.</param>
+        /// <param name="user">(Optional) Claims representing a user. This is useful on platforms like Blazor
+        /// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
+        /// will find the user from the HTTP request context.</param>
         /// <param name="content">Content to send to the REST API in the case where
-        /// <see cref="DownstreamRestApiOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
+        /// <see cref="AuthorizationHeaderProviderOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
         /// <see cref="HttpMethod.Post"/>, <see cref="HttpMethod.Put"/>.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
-        Task<HttpResponseMessage> CallRestApiForAppAsync(
+        Task<HttpResponseMessage> CallRestApiForUserAsync(
```

Comment:
How is this different than CallRestApiAsync?
Given there is a CallRestApiAsync, CallRestApiForUserAsync and CallRestApiForAppAsync how are these different?
If different actions are expected between CallRestApiForUserAsync and CallRestApiForAppAsync how will the implementation of CallRestApiAsync know what to do?